node/node.go

```diff
@@ -157,6 +158,13 @@ func NewNode(ctx context.Context, conf config.NodeConfig, p2pKey crypto.PrivKey,
 	if settlementlc == nil {
 		return nil, fmt.Errorf("couldn't get settlement client named '%s'", conf.SettlementLayer)
 	}
+	if conf.SettlementLayer == "mock" {
```

Comment:
do we have a way to make sure that making this check against mock is not done in our non mock related code? seems a bit dirty to me.